platform-dom/src/main/java/com/softicar/platform/dom/elements/testing/node/tester/AbstractDomNodeTester.java

```diff
@@ -86,27 +88,34 @@ public String getInputValue() {
 
 	// ------------------------------ events ------------------------------ //
 
-	public void sendEvent(DomEventType type) {
+	public AbstractDomNodeTester<N> sendEvent(DomEventType type) {
 
+		Assert.assertFalse("Trying to send %s event to disabled node.".formatted(type), isNodeDisabled());
 		engine.sendEvent(node, type);
+		return this;
 	}
 
 	public AbstractDomNodeTester<N> click() {
 
-		engine.sendEvent(node, DomEventType.CLICK);
-		return this;
+		return sendEvent(DomEventType.CLICK);
 	}
 
 	public AbstractDomNodeTester<N> doubleClick() {
 
-		engine.sendEvent(node, DomEventType.DBLCLICK);
-		return this;
+		return sendEvent(DomEventType.DBLCLICK);
 	}
 
 	public AbstractDomNodeTester<N> rightClick() {
 
-		engine.sendEvent(node, DomEventType.CONTEXTMENU);
-		return this;
+		return sendEvent(DomEventType.CONTEXTMENU);
+	}
+
+	private boolean isNodeDisabled() {
+
+		return CastUtils//
+			.tryCast(node, DomButton.class)
+			.map(button -> !button.isEnabled())
```

Comment:
`isNodeDisabled` _only_ checking the state for buttons (while ignoring any other element) is quite surprising.
I can see why it's that way, though.
In the future, we should probably add `isEnabled` to `IDomEnableable`, and make more widespread use of that interface.

Reply:
agreed, though since we merged anchors into buttons, this should cover about 99% of all cases